gcloud/rest/auth/iam.py

```diff
@@ -0,0 +1,149 @@
+import json
+import threading
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union  # pylint: disable=unused-import
+
+import requests
+
+from .token import Token
+from .token import Type
+from .utils import encode
+
+
+API_ROOT_IAM = 'https://iam.googleapis.com/v1'
+API_ROOT_IAM_CREDENTIALS = 'https://iamcredentials.googleapis.com/v1'
+SCOPES = ['https://www.googleapis.com/auth/iam']
+
+
+class IamClient:
+    def __init__(self,
+                 service_file=None,  # type: Optional[str]
+                 session=None,  # type: requests.Session
+                 google_api_lock=None,  # type: threading.RLock
+                 token=None  # type: Token
+                 ):
+        # type: (...) -> None
+        self.session = session
+        self.google_api_lock = google_api_lock or threading.RLock()
+        self.token = token or Token(service_file=service_file,
+                                    session=session, scopes=SCOPES)
+
+        if self.token.token_type != Type.SERVICE_ACCOUNT:
+            raise TypeError('IAM Credentials Client is only valid for use'
+                            ' with Service Accounts')
+
+    def headers(self):
+        # type: () -> Dict[str, str]
+        token = self.token.get()
+        return {
+            'Authorization': 'Bearer {}'.format(token),
+        }
+
+    @property
+    def service_account_email(self):
+        # type: () -> Optional[str]
+        return self.token.service_data.get('client_email')
+
+    # https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts.keys/get
+    # pylint: disable=too-many-arguments
```

Comment:
Let's just throw this into our `.pre-commit-config.yaml`

Reply:
will do